pkg/apis/pipeline/v1alpha1/cloud_event_resource.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2019 The Tekton Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// CloudEventResource is an event sink to which events are delivered when a TaskRun has finished
+type CloudEventResource struct {
+	Name      string               `json:"name"`
+	Type      PipelineResourceType `json:"type"`
+	TargetURI string               `json:"targetURI"`
+	Owner     TaskRun              `json:"taskRun"`
+}
+
+// NewCloudEventResource creates a new CloudEvent resource to pass to a Task
+func NewCloudEventResource(r *PipelineResource) (*CloudEventResource, error) {
```

Comment:
could be nice to include some test coverage of this function? basically just some simple tests to hit some of the if statements + loop logic :D

Reply:
done